### DISCUSSION TITLE
Permalink for module according to two fields

### DISCUSSION THREAD

**Discussion by goeil** [area17/twill#1247] [Q&A]
Hello,
Thanks a lot for Twill; I love it!

Is there any possibility to customize the slug string, accordining to two fields?
My desired example would be :

1. Have my two fields (e.g. First name and Last name) on the create modal.
2. Modifiy the slug in the same modal with <firstnameslug>-<lastnameslug> (e.g. 'peter-falk')

This is what I tried:
1. Customize the **create.blade.php** ; I managed to do this by overriding this file and adding my second field
2. Customize the formatPermalink function, but cannot find where

Thanks for your help

**Comment by pboivin** [comment]
Hi @goeil,

The short answer is yes, this is possible. However, it's not straightforward. Here are a few notes on how to approach this:

You will need a custom build of Twill's frontend, instead of the built-in one (`php artisan twill:build` instead of `twill:update`)

The `formatPermalink` function is defined in a Vue.js mixin: [formatPermalink.js](https://github.com/area17/twill/blob/2.x/frontend/js/mixins/formatPermalink.js). This mixin is used on both [main-listing.js](https://github.com/area17/twill/blob/2.x/frontend/js/main-listing.js#L37) and [main-form.js](https://github.com/area17/twill/blob/2.x/frontend/js/main-form.js#L57) root components.

So to customize this file as part of your custom build, you will need to override it in your project. I can think of 2 ways to do this:
- you can fork the twill repository on GitHub, make your changes there and use your fork instead of the main twill package
- a simpler option would be to copy your modified file into `vendor/area17/twill/frontend/js/mixins/` as part of your build step, before running `php artisan twill:build`

I haven't tested this extensively but this should work to compose a permalink out of 2 separate fields (non translated) :
```js
    formatNamePermalink: function () {
      const permalinkRef = this.$refs.permalink

      if (!permalinkRef) return

      const firstname = this.$store.getters.fieldValueByName('firstname') || ''
      const lastname = this.$store.getters.fieldValueByName('lastname') || ''
      const slug = this.$options.filters.slugify(`${firstname} ${lastname}`)

      const field = {
        name: permalinkRef.attributes ? permalinkRef.attributes.name : permalinkRef.name,
        value: slug
      }

      field.locale = this.currentLocale.value

      this.$store.commit(FORM.UPDATE_FORM_FIELD, field)
    }
```

You can then use this function in your custom `create.blade.php` template:
```blade
@formField('input', [
    'name' => 'firstname',
    'label' => 'First Name',
    'translated' => false,
    'required' => true,
    'onChange' => 'formatNamePermalink'
])

@formField('input', [
    'name' => 'lastname',
    'label' => 'Last Name',
    'translated' => false,
    'required' => true,
    'onChange' => 'formatNamePermalink'
])
```

Hope this can help!

**Reply by goeil** [reply]
So cool
thank you, it works as you gently indicated me.
Very promising.
